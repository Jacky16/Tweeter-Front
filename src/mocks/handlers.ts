// src/mocks/handlers.js
import { rest } from "msw";
import requestsUrl from "../config/requestsUrl";

export const handlers = [
  rest.post(requestsUrl.registerUser, (req, res, ctx) => {
    return res(ctx.status(201), ctx.json({}));
  }),
];
