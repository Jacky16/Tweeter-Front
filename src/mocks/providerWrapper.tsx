import { ThemeProvider } from "@mui/system";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../redux/store";
import mainTheme from "../styles/mainTheme";

interface ProviderWrapperProps {
  children: JSX.Element | JSX.Element[];
}
const ProviderWrapper = ({ children }: ProviderWrapperProps) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={mainTheme}>
        <Provider store={store}>{children}</Provider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default ProviderWrapper;
