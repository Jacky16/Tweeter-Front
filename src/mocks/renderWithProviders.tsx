import { configureStore, PreloadedState } from "@reduxjs/toolkit";
import { render, RenderOptions } from "@testing-library/react";
import { PropsWithChildren } from "react";
import { Provider } from "react-redux";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import { RootState, store } from "../redux/store";
import { tweetsReducer } from "../redux/tweetsSlice/tweetsSlice";
import { uiReducer } from "../redux/UiSlice/UiSlice";
import { userReducer } from "../redux/userSlice/userSlice";
import { InitialEntry } from "@remix-run/router";
interface ExtendedPropsWithChildren extends PropsWithChildren {
  initialEntries?: InitialEntry[];
}
interface ExtendedRenderOptions
  extends Omit<RenderOptions, "queries">,
    ExtendedPropsWithChildren {
  preloadedState?: PreloadedState<RootState>;
  store?: typeof store;
}

const Router = ({
  children,
  initialEntries,
}: ExtendedPropsWithChildren): JSX.Element => {
  return initialEntries ? (
    <MemoryRouter initialEntries={initialEntries}>{children}</MemoryRouter>
  ) : (
    <BrowserRouter>{children}</BrowserRouter>
  );
};

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    initialEntries,
    preloadedState,
    store = configureStore({
      reducer: { ui: uiReducer, user: userReducer, tweets: tweetsReducer },
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <Router initialEntries={initialEntries}>
        <Provider store={store}>{children}</Provider>
      </Router>
    );
  };
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};
