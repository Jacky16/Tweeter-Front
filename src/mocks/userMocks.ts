import { UserState } from "../redux/types";

export const mockUser: UserState = {
  isLogged: false,
  token: "abc",
  id: "123",
  alias: "@mario",
  username: "mario",
};
