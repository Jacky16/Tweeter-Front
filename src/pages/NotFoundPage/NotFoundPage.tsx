import { Link as RouterLink } from "react-router-dom";
import { Stack, Typography, Link } from "@mui/material";

const NotFoundPage = () => {
  const fontWeight = 900;
  return (
    <Stack justifyContent={"center"} direction={"column"} alignItems={"center"}>
      <Typography fontSize={100} fontWeight={fontWeight}>
        404
      </Typography>
      <Typography fontSize={65} fontWeight={fontWeight} textAlign={"center"}>
        THIS PAGE DOESNâ€™T EXISTS
      </Typography>
      <Typography fontWeight={fontWeight} fontSize={40} textAlign={"center"}>
        Go to{" "}
        <Link component={RouterLink} to={"/"}>
          home
        </Link>
      </Typography>
    </Stack>
  );
};

export default NotFoundPage;
