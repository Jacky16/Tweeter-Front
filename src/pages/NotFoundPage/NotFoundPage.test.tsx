import { renderWithProviders } from "../../mocks/renderWithProviders";
import NotFoundPage from "./NotFoundPage";

describe("Given the NotFoundPage", () => {
  describe("When it is rendered", () => {
    test("Then it should match the snapshot", () => {
      const view = renderWithProviders(<NotFoundPage />);
      expect(view).toMatchSnapshot();
    });
  });
});
