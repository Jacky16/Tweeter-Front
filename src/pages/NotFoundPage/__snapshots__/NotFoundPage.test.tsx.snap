// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the NotFoundPage When it is rendered Then it should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-w004l2-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1y2wcwp-MuiTypography-root"
        >
          404
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1bbgn34-MuiTypography-root"
        >
          THIS PAGE DOESN’T EXISTS
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-r2cpg4-MuiTypography-root"
        >
          Go to
           
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            href="/"
          >
            home
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-w004l2-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1y2wcwp-MuiTypography-root"
      >
        404
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1bbgn34-MuiTypography-root"
      >
        THIS PAGE DOESN’T EXISTS
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-r2cpg4-MuiTypography-root"
      >
        Go to
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
          href="/"
        >
          home
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
